src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -343,6 +346,30 @@ request_creator::make_delete_objects_request(
         std::make_unique<delete_objects_body>(std::move(body))}}};
 }
 
+std::string
+request_creator::make_host([[maybe_unused]] const bucket_name& name) const {
```

Comment:
you do don't need `maybe_unused` -- it's only needed if the code generated doesn't use it (e.g. in conjunction with constexpr or templates etc...). in this case it is used in the virtual host case.